pkg/analysis/helpers/markers/analyzer.go

```diff
@@ -167,18 +167,80 @@ func extractMarker(comment *ast.Comment) Marker {
 		return Marker{}
 	}
 
+	markerContent := strings.TrimPrefix(comment.Text, "// +")
+	id, expressions := extractMarkerIdAndExpressions(markerContent)
+
 	return Marker{
-		Value:      strings.TrimPrefix(comment.Text, "// +"),
-		RawComment: comment.Text,
-		Pos:        comment.Pos(),
-		End:        comment.End(),
+		Identifier:  id,
+		Expressions: expressions,
+		RawComment:  comment.Text,
+		Pos:         comment.Pos(),
+		End:         comment.End(),
+	}
+}
+
+func extractMarkerIdAndExpressions(marker string) (string, map[string]string) {
+	// if there is only a single "=" split on the equal sign and trim any
+	// dangling ":" characters.
+	if strings.Count(marker, "=") == 1 {
+		splits := strings.Split(marker, "=")
+		// Trim any dangling ":" characters on the identifier to handle
+		// cases like +kubebuilder:object:root:=true
+		identifier := strings.TrimSuffix(splits[0], ":")
+
+		// If there is a single "=" sign that means the left side of the
+		// marker is the identifier and there is no real expression identifier.
+		// For now, set the expression identifier as an empty string.
+		// TODO: Is this sufficient? Can this be improved?
+		expressions := map[string]string{"": splits[1]}
+		return identifier, expressions
+	}
+
+	// split on :
+	separators := strings.Split(marker, ":")
+	// Means no : characters. Probably a marker like +default, +optional, etc.
+	if len(separators) == 1 {
+		return separators[0], map[string]string{}
+	}
```

Comment:
Is this actually required? I feel like falling into the logic below would end up with the same result?

Reply:
I think you are right. I don't quite recall why I made this separate, but knowing myself it was probably to have a distinct early return while I was still deciding if this was the right logic and I didn't do a good enough job of reviewing my own code post-recharge before marking this as ready :P - good catch :)